netbox/utilities/forms/bulk_import.py

```diff
@@ -90,8 +90,17 @@ def _clean_csv(self, data):
         """
         Clean CSV-formatted data. The first row will be treated as column headers.
         """
+
+        # Determine the CSV dialect
+        try:
+            # This uses a rough heuristic to detect the CSV dialect. If the data is malformed, we'll fall back to
+            # the default Excel dialect.
+            dialect = csv.Sniffer().sniff(data.strip(), delimiters=',; ')
```

Comment:
Should `delimiters` include a space? Should this be a tab (`\t`) instead?

Reply:
I tried but the internals for sniff simply iterate over the single char like `[key for key in delimiters]` due to which it's not really possible to pass `\t`. I tried using a list of strings but my IDE was telling me the expected type is `str` or `None` only,

Reply:
Maybe I'm misunderstanding, but seems to work here?

```
>>> data = """test1\ttest2\ttest3
... dss\tdsds\tdsds"""
>>> print(data)
test1   test2   test3
dss     dsds    dsds
>>> dialect = csv.Sniffer().sniff(data.strip(), delimiters=',;\t')
>>> dialect.delimiter
'\t'
```

Also:

```
>>> [key for key in ",;\t"]
[',', ';', '\t']
```

Reply:
This is odd, when I tried from the Import UI, the code blew up because it could not detect the delimiter

Reply:
It's my fault I guess to assume the issue originated at the sniff function, rather it was with header checks

![image](https://github.com/netbox-community/netbox/assets/5083532/9de59190-eb8e-4fed-96cb-01e0ffe64350)

Reply:
This screenshot makes no sense, you're trying to put \t into the web form? That's not the same as a tab character.

Try creating a text file in Notepad with tabs seperating the fields and copy-paste this into the form instead.